src/ocp_postprocess.rs

```diff
@@ -292,14 +422,29 @@ async fn fix_dep_annotations(
             "v1",
         );
 
-        let data_json = &serde_json::to_string(
-            get_etcd_json(etcd_client, &resource_k8s_resource_location)
-                .await?
-                .context(format!("couldn't find {}", resource_k8s_resource_location))?
-                .pointer("/data")
-                .context("no .data")?,
-        )
-        .context("couldn't serialize data")?;
+        let data = get_etcd_json(etcd_client, &resource_k8s_resource_location)
+            .await?
+            .context(format!("couldn't find {}", resource_k8s_resource_location))?
+            .pointer("/data")
+            .context("no .data")?
+            .as_object()
+            .context("data not an object")?
+            .clone();
+
+        let data_json = if resource_k8s_resource_location.kind == "secret" {
+            // https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/encoding/json/encode.go;l=751-753
+            // https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/encoding/json/encode.go;l=790
+            let sorted: BTreeMap<_, _> = data
+                .iter()
+                .map(|(k, v)| (k, base64_standard.encode(serde_json::from_value::<Vec<u8>>(v.clone()).unwrap())))
```

Comment:
no unwrapping

Reply:
Any ideas on how to get rid of this `unwrap`? I've been unsuccessfully playing around with it for a bit.

Reply:
Return a vector of results from the map, instead of a vector of strings

A vector of `Result<T>`'s knows how to be collected into a `Result<Vec<T>>`


```rust
        let data_json = if resource_k8s_resource_location.kind == "secret" {
            // https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/encoding/json/encode.go;l=751-753
            // https://cs.opensource.google/go/go/+/refs/tags/go1.22.1:src/encoding/json/encode.go;l=790
            let sorted: BTreeMap<_, _> = data
                .iter()
                .map(|(k, v)| {
                    Ok((
                        k,
                        base64_standard.encode(serde_json::from_value::<Vec<u8>>(v.clone()).context("oops")?),
                    ))
                })
                .collect::<Result<Vec<_>>>()?
                .into_iter()
                .collect();
```

Reply:
Thank you for the detailed explanation!